### PR TITLE
feat: added embeds with author

### PR DESCRIPTION
**Pull request by dwight-schrute**
- removed import dependency outside of ii repos
- embed addition / modification for weighted / list commands
- fetching description for all results for weighted command (due to multiple network calls, the command is getting timed out, have to fix this)

### CODE REVIEW

**Comment by dwight-schrute** [comment]
I accept the CLA

**Comment by crimelabs786** [comment]
Thanks @dwight-schrute for accepting CLA, have added you to list of CLA verified authors

@cla-bot check now

File: gitbook_client.ts [hunk 1]
```diff
@@ -78,11 +78,13 @@ export class GitbookSpaceClient {
   }
 
   async searchSpace(query: string) {
-    const { results } = (await this.get(
+    const startTime = new Date().getTime();
```
Comment (crimelabs786) [inline gitbook_client.ts:81]: `Date.now()` is better

File: gitbook_client.ts [hunk 2]
```diff
@@ -78,11 +78,13 @@ export class GitbookSpaceClient {
...
+    let { results } = (await this.get(
       "search",
       new URLSearchParams({ query })
     )) as { results: GitbookSearchNode[] };
-    return results
+    const timeTaken = (new Date().getTime() - startTime) / (1000);
```
Comment (crimelabs786) [inline gitbook_client.ts:86]: Same as [above](#r633265137)

File: gitbook_commands.ts [hunk 3]
```diff
@@ -88,37 +114,45 @@ slash.registerHandler("weighted", async (interaction) => {
...
+    }
 
+    if (desc && desc !== "") {
       const em = new Embed().setDescription(desc).setColor(color);
-
       embeds.push(em);
     }
-
+    
+    embeds[embeds.length - 1].setFooter(`\/list query: ${query.value} | retrieved in ${(timeTaken)} seconds`)
```
Comment (crimelabs786) [inline gitbook_commands.ts:155]: Don't hardcode the command name 😉

**Review by crimelabs786 — APPROVED** [review]

**Pull request merged**